admin_guide/sdn_troubleshooting.adoc

```diff
@@ -0,0 +1,527 @@
+define your
+link:architecture/core_concepts/pods_and_services.html[service].
+
+The IP addresses listed in the service output are the Kubernetes
+service IP addresses that Kubernetes will map to one of the pods that
+backs that service.  So you should be able to talk to that IP address.
+But, unfortunately, even if you can, it doesn't mean all pods are reachable;
+and if you can't, it doesn't mean all pods aren't reachable.  It just
+tells you the status of the /one/ that kubproxy hooked you up to.
+
+Let's test the service anyway:
+....
+curl -kv http://172.30.243.225:5000/bar                  # Replace the argument with your service IP address and port
+....
+
+Then, let's work out what pods are backing our service (replace `docker-registry` with the name of the broken service):
+....
+# oc get endpoints --selector=docker-registry
+NAME              ENDPOINTS
+docker-registry   10.1.2.2:5000
+....
+
+From this, we can see that there's only one endpoint.  So, if your
+service test succeeded, and the router test succeeded, then something
+really odd is going on.  But if there's more than one endpoint, or the
+service test failed, try the following _for each_ endpoint.  Once you
+identify what endpoints aren't working, then proceed to the next
+section.
+
+First, test each endpoint (change the URL to have the right endpoint IP, port, and path):
+....
+curl -kv http://10.1.2.2:5000/bar
+....
+
+If that works, great, try the next one.  If it failed, make a note of
+it and we'll work out why, in the next section.
+
+If all of them failed, then it is possible that the local node is not
+working, jump to the <<debugging-local-networking>> section.
+
+If all of them worked, then jump to the <<debugging-kubernetes>>
+section to work out why the service IP address isn't working.
+
+
+== Debugging Node to Node Networking
+
+Using our list of non-working endpoints, we need to test connectivitiy
+to the node first.  First we need to get a list of the endpoint IPs
+and node IPs:
+....
+# oc get pods --selector=docker-registry --template='{{range .items}}HostIP: {{.status.hostIP}}   PodIP: {{.status.podIP}}{{end}}{{"\n"}}'  # Again, replace docker registry with the one you are interested in
+HostIP: 192.168.122.202   PodIP: 10.1.0.4
+....
+
+Then find the endpoint IP address you made note of before and look for
+it in the PodIP column, and find the corresponding Host IP address.
+Then test connectivity at the node (host) level (using the address from
+`HostIP`):
+
+* `ping -c 3 _address_`: This should show a response, but if it doesn't it is possible that an intermediate router is eating the ICMP traffic.
+* `tracepath _address_`:  If both `tracepath` and `ping` failed, then you need to look for connectivity issues with your local network (or any containing VMs, if any).
+
+Things to check for the local networking:
+
+* `ip route get _address_`: Prints the IP route taken (if ICMP packets are returned by all hops)
```

Comment:
Whoops... I had more description of the tracepath, and then mangled something in an edit.